.github/workflows/ci.yml

```diff
@@ -207,30 +207,6 @@ jobs:
           path: ${{ matrix.artifact-path }}
           if-no-files-found: error
 
-  linux-cmake:
-    name: 🐧 Build (Linux, GCC, CMake)
-    runs-on: ubuntu-22.04
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-        with:
-          submodules: recursive
-
-      - name: Install dependencies
-        run: |
-          sudo apt-get update -qq
-          sudo apt-get install -qqq build-essential pkg-config cmake
-
-      - name: Build godot-cpp
-        run: |
-          cmake -DCMAKE_BUILD_TYPE=Release .
-          make -j $(nproc) VERBOSE=1
-
-      - name: Build test GDExtension library
-        run: |
-          cd test && cmake -DCMAKE_BUILD_TYPE=Release -DGODOT_HEADERS_PATH="../godot-headers" -DCPP_BINDINGS_PATH=".." .
-          make -j $(nproc) VERBOSE=1
```

Comment:
Ninja builds significantly faster and it's easy to install on CI, so I would recommend that instead.

Reply:
When looking into the contents of makefiles and ninja files they are so similar as to seem a negligible difference in structure. I would highly doubt that there would be a way to make them conflict unless you were doing something ultra weird. I also dont think that it's necessary to try to maximise coverage of the available options, as there are a 
[bunch](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html).

So far I'm testing on commandline for: Visual Studio 2022, Ninja, and XCode (I have to put more effort into macos and XCode) which covers all of the platforms and compilers I can get my hands on so far. I think it's unnecessary overhead to support more than these three. But if you insist I will bring it back. I Am planning on going over @IvanInventor 's CMake CI #1452 and getting that updated and merged with his help.

What do you think? do we really need to cover more than one cmake generated build solution per platform in the CI? Perhaps we can have a seperate task for maximum coverage we can run periodically rather than every PR with all options?

Reply:
@enetheru You're the CMake expert, I'll trust you! If you're confident that using CMake with the "UNIX Makefiles" generator will work for users if we've tested that the "Ninja" generator is working, then I'm OK with dropping this job. (And if we do encounter issues later on, we can always add it back.)